### PR TITLE
Kubernetes recommanded labels

### PR DESCRIPTION
**Pull request by nlamirault**
Next of #1538.

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by STRRL — CHANGES_REQUESTED** [review]
Thanks for your contribution! That's really hard work!

I noticed that the generated install.sh have extra 4 whitespace after the `label:`, I think we could fix that by using `nindent 4` instead of `indent 4`. And the `{{- define "chaos-mesh.labels" }}` in `_helpers.tpl` should be changed to `{{- define "chaos-mesh.labels" -}}`, with a suffix `-` to trim the whitespace/line-wrap.

And you might miss two place to use `{{ include template }}`, the one is the Service of chaos-dashboard at helm/chaos-mesh/templates/chaos-dashboard-deployment.yaml:88, the other one is ValidatingWebhookConfiguration at helm/chaos-mesh/templates/secrets-configuration.yaml:172

**Comment by STRRL** [comment]
Hi @nlamirault, is this PR ready for review? 😁

**Comment by nlamirault** [comment]
@STRRL yes

**Comment by STRRL** [comment]
> And you might miss two place to use {{ include template }}, the one is the Service of chaos-dashboard at helm/chaos-mesh/templates/chaos-dashboard-deployment.yaml:88, the other one is ValidatingWebhookConfiguration at helm/chaos-mesh/templates/secrets-configuration.yaml:172

Hi @nlamirault, I think you might miss these two places, it could be instead with template "chaos-mesh.labels".

File: helm/chaos-mesh/templates/_helpers.tpl [hunk 1]
```diff
@@ -31,6 +31,27 @@ Create chart name and version as used by the chart label.
...
+{{- end }}
+{{- end }}
+
+{{/*
+Specify default selectors
+*/}}
+{{- define "chaos-mesh.selectors" }}
+app.kubernetes.io/name: {{ template "chaos-mesh.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
```
Comment (STRRL) [inline helm/chaos-mesh/templates/_helpers.tpl:53]: I think we could make the same change like "chaos-mesh.labels" for preventing extra whitespace. :D

**Review by STRRL — CHANGES_REQUESTED** [review]
rest LGTM

**Comment by nlamirault** [comment]
I will fixe that asap

**Review by STRRL — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
Hi @nlamirault , I saw that the e2e test failed because it failed to install chaos-mesh by helm.

I think you should use `include "chaos-mesh.selectors" .` and `include "chaos-mesh.labels" .` with `nindent`, not `indent`, that will fix issue.

**Review by cwen0 — APPROVED** [review]
LGTM

**Review by STRRL — APPROVED** [review]
LGTM

**Pull request merged**